src/main/java/soot/jimple/validation/FieldRefValidator.java

```diff
@@ -66,26 +66,25 @@ public void validate(Body body, List<ValidationException> exceptions) {
         try {
           SootField field = v.getField();
           if (field == null) {
-            exceptions.add(new UnitValidationException(unit, body, "Resolved field is null: " + fr.toString()));
+            exceptions.add(new UnitValidationException(u, body, "Resolved field is null: " + fr.toString()));
           } else if (!field.isStatic() && !field.isPhantom()) {
-            exceptions
-                .add(new UnitValidationException(unit, body, "Trying to get a static field which is non-static: " + v));
+            exceptions.add(new UnitValidationException(u, body, "Used StaticFieldRef for a non-static field: " + v));
           }
         } catch (ResolutionFailedException e) {
-          exceptions.add(new UnitValidationException(unit, body, "Trying to get a static field which is non-static: " + v));
+          exceptions.add(new UnitValidationException(u, body, "Unable to resolve FieldRef " + v + ": " + e.getMessage()));
```

Comment:
Doesn't the exception happen when resolving a StaticFieldRef? Similarly, at line 87 it happens when trying to resolve an InstanceFieldRef.

Reply:
Yes, that's obvious from the code. The problem is, a mismatch of static vs non-static is not the reason that a ```ResolutionFailedException``` can occur so the message is misleading. If you look at ```AbstractSootFieldRef.java``` you will also see that ```resolve()``` could also throw ```FieldResolutionFailedException``` (a subclass of ```ResolutionFailedException```) which has nothing to do with the static-ness of the field. That is the reason I replaced the majority of the String formed here with the message from the exception itself because that will provide more specific information.

Reply:
If you think it's better, I could change two message to "Unable to resolve StaticFieldRef" and "Unable to resolve InstanceFieldRef" respectively. That can provide a little extra information (although it's something that looking at the stack trace of the exception would already tell you).